### ISSUE TITLE
aws_lru_cache undefined?

### ISSUE THREAD

**Issue by eddelbuettel** [awslabs/aws-c-common#668]
A little puzzling, and I am not entirely sure I am not overlooking something entirely trivial, but when trying to build and test this repo in isolation, it stumbles over one test file `tests/lru_cache_test.c` 

```
/home/ubuntu/git/pkg-aws-c-common/pkg/tests/lru_cache_test.c: In function ‘s_test_lru_cache_overflow_static_members_fn’:
/home/ubuntu/git/pkg-aws-c-common/pkg/tests/lru_cache_test.c:22:26: error: storage size of ‘cache’ isn’t known     
     struct aws_lru_cache cache;                                                                         
                          ^~~~~                                                                          
In file included from /home/ubuntu/git/pkg-aws-c-common/pkg/tests/lru_cache_test.c:17:0:             
```

And a quick recursive `grep` confirms that this appears to be undefined.   I am not entirely sure how that could pass your CI then.  So for now and my use I just disabled the test file.

Verified on two machines with Ubuntu 18.04 and 20.04 both running a checkout just before the very recent commit minutes ago.

**Comment by bretambrose** [comment]
lru_cache_test.c has not existed since this commit here a couple months ago: https://github.com/awslabs/aws-c-common/commit/17e0a13b1cbee65ac21ddf4f123e38d89b0fd596

It seems like you somehow have a weird state that is mixing old and new versions.  I don't recognize the /git/pkg-* directory structure.  How are you pulling down the source?

**Comment by eddelbuettel** [comment]
Bingo---likely my mistake then.  I must have rsync'ed without the `--delete` option -- my bad!  

[ Context:  What I am doing is slightly different from just `git pull; cd build; cmake ..` as I am building (local, but also built in PPA at launchpad) .deb packages to support the subset of the C++ SDK I need for s3 support.  I have all I do up here on github under my name and repo name pattern pkg-aws-* ]  

So in short please do feel free to close at your leisure, I can too -- and many thanks for the very prompt response.

**Comment by eddelbuettel** [comment]
Confirming that `rsync -cav --delete --exclude debian/ --exclude build/ --exclude .git` removes the file in question. I feel like a fool now.  Sorry for the noise.

**Comment by bretambrose** [comment]
Don't say that!  Quick resolutions are the highlight of my oncall.  =D
